### PR TITLE
[Fix] `ExportMap` / flat config: include `languageOptions` in context

### LINKED ISSUE #3051: Missing documentation on how to setup eslint-plugin-import with flat confiuration and Typescript
Readme only contains information about how to setup typescript for this plugin using the old (non-flat) eslint configuration format.
https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#typescript

Upgraded to latest eslint-plugin-import and now the compat functions from @eslint/compat cant be used any longer (fixupConfigRules/fixupPluginRules) and i don't manage to lint my ts files any longer.

I get loads of errors like this:

```
 error  Parse errors in imported module '@emotion/cache': parserPath or languageOptions.parser is required! (undefined:undefined)  import/default
 ```
 
 I have a languageOptions.parser defined in my config:
 
 ```
 const tsParser = require('@typescript-eslint/parser');
 ...
   
 languageOptions: {
            parser: tsParser,
            ecmaVersion: 6,
            sourceType: 'module',
            parserOptions: {
                ecmaFeatures: {
                    jsx: true,
                },
            },
        },
```

### PR DESCRIPTION
**Pull request by michaelfaith**
This change fixes a bug with flat config support.  There is a function called `childContext` that's used by the ExportBuilder to "cleanse" the context object.  This function wasn't including the new `languageOptions` object, which contains the parser.  So by the time this cleansed context made it to the parse function, `languageOptions` wasn't there anymore.

Since `parserPath` was still being included in non-flat config scenarios, the parse function made it through ok and used `parserPath`.  However, once you shift to flat config, `parserPath` is no longer defined, and the actual parser object needs to be there.

Fixes #3051

### CODE REVIEW

**Review by ljharb — COMMENTED** [review]
Thanks! Any way we can test for this?

**Comment by michaelfaith** [comment]
> Thanks! Any way we can test for this?

For sure.  I can do that.  I'm having issues running the unit test suite locally (I think my version of node is too new for this version of `nyc`), so it may take a few updates to the PR to get things just right, using CI as my feedback loop.

**Comment by michaelfaith** [comment]
Added a couple of tests for `childContext` (and I was able to get the tests running locally by updating the version of `nyc` to the latest, as well as making a few other tweaks to the setup).

**Comment by rarenatoe** [comment]
Do you guys think this will get merged and released today?

**Review by ljharb — APPROVED** [review]
Thanks! It'd be really nice to have an actual eslint 9 test that matches the flow that exposed this issue, but this is sufficient for now :-)

**Comment by michaelfaith** [comment]
> Thanks! It'd be really nice to have an actual eslint 9 test that matches the flow that exposed this issue, but this is sufficient for now :-)

Totally agree, and those'll be in @G-Rath's PR.  Their v9 tests are failing right now because they need this change.  So, once they bring this into their branch, those tests will be the v9 integration tests that cover this scenario.

**Comment by G-Rath** [comment]
(fwiw I did a poor cherry-pick before this landed and pushed that up and at least some of the tests were improved, and now locally I've rebased and am just sorting out the handling of `env` before I push again and start digging into the whole failures etc etc 🙂)

**Comment by ljharb** [comment]
I'm waiting for CI, and then I'll be publishing eslint-module-utils, and then updating `main` to require it and use the new helpers, after which you'll need to rebase, so feel free to wait :-)

**Pull request merged**
